addons/hr_holidays_attendance/models/hr_leave.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from odoo.tools import float_round
+
+
+class HRLeave(models.Model):
+    _inherit = 'hr.leave'
+
+    overtime_id = fields.Many2one('hr.attendance.overtime', string='Extra Hours', groups='hr_holidays.group_hr_holidays_user')
+    employee_overtime = fields.Float(related='employee_id.total_overtime')
+    overtime_deductible = fields.Boolean(compute='_compute_overtime_deductible')
+
+    @api.depends('holiday_status_id')
+    def _compute_overtime_deductible(self):
+        self.overtime_deductible = False
+        deductibles = self.filtered(lambda l: l.holiday_status_id.overtime_deductible and l.holiday_status_id.allocation_type == 'no')
+        deductibles.overtime_deductible = True
+
+    @api.model
+    def create(self, vals):
+        leave_type = self.env['hr.leave.type'].browse(vals.get('holiday_status_id'))
+        if leave_type.overtime_deductible and leave_type.allocation_type == 'no':
+            employee_sudo = self.env['hr.employee'].sudo().browse(vals.get('employee_id'))
+            calendar = employee_sudo.resource_calendar_id or self.env.company.resource_calendar_id
+            date_from = fields.Datetime.to_datetime(vals['date_from'])
+            date_to = fields.Datetime.to_datetime(vals['date_to'])
+            duration = self._get_number_of_hours_display(
+                date_from, date_to,
+                vals['number_of_days'], vals['state'] if 'state' in vals else 'draft',
+                employee_sudo, calendar,
+            )
+            if duration > employee_sudo.total_overtime:
+                raise ValidationError(_('The employee does not have enough extra hours to request this leave.'))
+
+            if not vals.get('overtime_id'):
+                overtime_id = self.env['hr.attendance.overtime'].sudo().create({
+                    'employee_id': employee_sudo.id,
+                    'date': fields.Date.today(),
+                    'adjustment': True,
+                    'duration': -1 * duration,
+                })
+                vals['overtime_id'] = overtime_id.id
+
+        return super().create(vals)
+
+    def write(self, vals):
+        if 'number_of_days' in vals:
+            num_days = vals['number_of_days']
+            employee_sudo = self.env['hr.employee'].sudo().browse(vals['employee_id']) if 'employee_id' in vals else None
+            for leave in self.filtered('overtime_id'):
+                employee = employee_sudo if employee_sudo else leave.employee_id
+                calendar = employee.resource_calendar_id or self.env.company.resource_calendar_id
```

Comment:
To avoid multi-company issues, you shouldn't use self.env.company

Reply:
Changed to:

`                calendar = employee.resource_calendar_id or employee.company_id.resource_calendar_id`